src/db.rs

```diff
@@ -259,8 +263,11 @@ impl DB {
     }
 
     fn get_path_tmp(&self) -> PathBuf {
+        let file_name = format!(".{}.zo", Uuid::new_v4());
```

Comment:
What problem is this solving? I personally don't like the idea of adding **8** new dependencies just to get a random filename.

Reply:
This was done in order to achieve thread-safety. Currently, there is no locking mechanism in place, which means multiple instances of `zoxide` could be writing to the same tmp file concurrently, which would corrupt the database.

My initial idea was to use `flock`s to ensure that only one transaction can take place at a time. However, that isn't a great idea, since a misbehaving `zoxide` instance would hang every shell instance on the system.

To prevent this, I'm now generating a new tmp filename for every transaction. This ensures that while some transactions may be lost, the database itself will never be corrupted.

The reason I've used UUIDs (and not an RNG) to generate the filename is because they are virtually guaranteed to be unique, which saves us the trouble of checking if there is already a tmp file by that name.

An alternative to this would be to use [`create_new`](https://doc.rust-lang.org/std/fs/struct.OpenOptions.html#method.create_new), which will ensure that we don't accidentally write to a preexisting tmp file. Whether or not we remove UUIDs, I think `create_new` would be a good thing to have here.

Reply:
Still not happy about it, but it does solve an actual problem. I'll look into lighter alternatives, but that shouldn't block this getting merged.

`create_new` will error if the file already exists, so while it's a good addition, it's not a solution. Looks like `uuid` will be necessary to address this concern, at least for now.

Reply:
I'll mark this as resolved, since `uuid` can be replaced painlessly in future releases. Do create an issue if you find a better way of doing this.